src/HttpHeader.cc

```diff
@@ -747,6 +748,22 @@ HttpHeader::refreshMask()
     }
 }
 
+bool
+HttpHeader::tooLarge() const
+{
+    size_t max = 0;
+    switch (owner) {
+    case hoRequest:
+        max = Config.maxRequestHeaderSize;
+    case hoReply:
+         max = Config.maxReplyHeaderSize;
+    default:
+        return false;
```

Comment:
Let's handle all cases so that future code modifications do not forget to handle a new case:

```suggestion
    case hoEnd:
         // this case should not happen
        return false;
```

or, if we are relatively confident,

```suggestion
    case hoEnd:
        Assure(!"bad HttpHeader owner");
```